app/services/files/create_service.rb

```diff
@@ -9,9 +9,14 @@ def commit
     def validate
       super
 
-      file_name = File.basename(@file_path)
+      if @file_path =~ Gitlab::Regex.directory_traversal_regex
```

Comment:
Can't we rescue from the `Repository::InvalidBlobName` error that Gitlab::Git raises, then?